http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/KubernetesServiceInterp.scala

```diff
@@ -255,15 +255,32 @@ final class LeoKubernetesServiceInterp[F[_]: Parallel](
         )(d => F.pure(Some(d.id)))
       else F.pure[Option[DiskId]](None)
 
-      deleteMessage = DeleteAppMessage(
-        appResult.app.id,
-        appResult.app.appName,
-        appResult.nodepool.id,
-        appResult.cluster.googleProject,
-        diskOpt,
-        Some(ctx.traceId)
-      )
-      _ <- publisherQueue.enqueue1(deleteMessage)
+      // If the app status is Error, we can assume that the underlying app/nodepool
+      // has already been deleted. So we just transition the app to Deleted status
+      // without sending a message to Back Leo.
+      //
+      // Note this has the side effect of not deleting the disk if requested to do so. The
+      // caller must manually delete the disk in this situation. We have the same behavior for
+      // runtimes.
```

Comment:
For the aforementioned side effect, if the user doesn't delete the disk, would any of the clean-up mechanisms delete it today? If not, would it make sense to either send a `delete disk` message or to create a cron job (as a separate ticket as needed) to detect this case and clean the disk up after the fact. It feels like it may be too easy for users to not be aware that they have to delete the disk separately, which would incur costs silently, which usually makes it important to address.

Reply:
I don't think anything would automatically delete the disk if this happens. In generate we never want to automatically delete disks unless the user explicitly requests it.

In the UI the user can view all the active disks they currently have, and the UI warns you if you have >1 disk at a time. So I think the user would be notified, but may be annoyed they have to separately delete the disk when they get an `Error`'d runtime or app. 

Sending a `delete disk` message in this situation may make sense, but I'd kind of prefer to ticket that separately since the bug exists for runtimes and apps and is not directly related to the bug this PR is trying to address.

Reply:
> I don't think anything would automatically delete the disk if this happens. In generate we never want to automatically delete disks unless the user explicitly requests it.

Certainly. To clarify, I was talking about the case where a user does request the disk to be deleted as part of the `deleteApp` request.

> Sending a delete disk message in this situation may make sense, but I'd kind of prefer to ticket that separately since the bug exists for runtimes and apps and is not directly related to the bug this PR is trying to address.

Creating a separate ticket to handle it for both apps and runtimes sounds good to me 😊